### ISSUE TITLE
`solve_cubic` returns NaN when Q == 0 and R < 0, e.g. x^3 + 1 = 0

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4832]
`solve_cubic`'s `Q == 0` branch takes the real cube root with `torch.pow(2 * R, 1 / 3)` (`kornia/geometry/solvers/polynomial_solver.py`, the `mask_Q_zero` block). `torch.pow` of a negative base to a fractional power is NaN, so every cubic whose depressed form is `t^3 = 2R` with `R < 0` returns NaN instead of its real root. This happens in every dtype.

## Repro (main `012997ceb`, torch 2.14.0, CPU)

```python
import torch
from kornia.geometry.solvers import solve_cubic, solve_quartic

for c in ([1.0, 0.0, 0.0, 1.0], [1.0, 0.0, 0.0, 8.0], [1.0, -7.5, 18.75, -15.5], [1.0, 0.0, 0.0, -1.0]):
    print(c, solve_cubic(torch.tensor([c], dtype=torch.float64))[0].tolist())
print(solve_quartic(torch.tensor([[0.0, 1.0, 0.0, 0.0, 1.0]], dtype=torch.float64)))
```

```
[1.0, 0.0, 0.0, 1.0] [nan, 0.0, 0.0]        # x^3 + 1: root -1
[1.0, 0.0, 0.0, 8.0] [nan, 0.0, 0.0]        # x^3 + 8: root -2
[1.0, -7.5, 18.75, -15.5] [nan, 0.0, 0.0]   # (x - 2)(x^2 - 5.5x + 7.75): root 2
[1.0, 0.0, 0.0, -1.0] [1.0, 0.0, 0.0]       # R > 0 is fine
tensor([[nan, 0., 0., 0.]], dtype=torch.float64)   # solve_quartic's cubic fallback inherits it
```

float32 gives the same result. In bfloat16, rounding can send `Q` to exactly 0 near a triple root. For example, cubics with roots near `3.22, 3.23, 2.85` round to `[1, -9.3125, 28.875, -29.75]`, and `solve_cubic` returns `[nan, 0, 0]`. 7 of 2000 random bfloat16 cubics with roots in `[-4, 4]` return NaN, and `Q` is exactly 0 for both rows I inspected.

## Expected

The real cube root: `-1`, `-2` and `2` above. `torch.sign(R) * torch.pow(2 * torch.abs(R), 1 / 3)` (the branch already requires `R != 0`) gives it. The gradient of `pow(|2R|, 1/3)` is finite there for the same reason. Pin `x^3 + 1` and the `-7.5` literal in float32/float64.

Found while reviewing #4822; unrelated to it.

Posted on behalf of @ducha-aiki by Claude (Opus 5.5).
